experimental/devices/hd44780/hd44780.go

```diff
@@ -0,0 +1,261 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package hd44780 controls the Hitachi LCD display chipset HD-44780
+//
+// Datasheet
+//
+// https://www.sparkfun.com/datasheets/LCD/HD44780.pdf
+package hd44780
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/gpio"
+)
+
+// lineTwo offset for the second line in the LCD buffer.
+const lineTwo = 0x40
+
+// Dev is the 4-bit addressing device for HD-44780
+type Dev struct {
+	// data pins
+	dataPins []gpio.PinOut
+
+	// register select pin
+	rsPin gpio.PinOut
+
+	// enable pin
+	enablePin gpio.PinOut
```

Comment:
I think the driver should keep track of the current cursor position, so it can skip updating the cursor when unnecessary.

You said the chip keep track of the current position, so you should leverage this by keeping state here too.

Reply:
The logic here is: you can use `SetCursor` to position the cursor at certain point, then you can call arbitrary number of `Print` to update the content of string. No need to track current cursor position inside the driver.

Additionally, you can use `PrintAt` to specify the exact location of the text to be printed on the screen. Again, I don't see a value to track the cursor inside the driver. If a client app needs to have something sophisticated - they can do it internally.